### PR TITLE
Return option

### PR DESCRIPTION
**Pull request by MirijaHa**
This PR adds the option to return st results in R instead of exporting them to the output directory. @jakubcervenka and @daisy-pacheco  might like this.

NOTE: I merely copy pasted code around and did not refactor any internals to gain implementation speed.

@jacobvjk 
I did not include the new option in the vignettes yet, would be great if you could do it in a follow up.

On change checks:

- checks passed for all asset types with and without iteration
- I compared also results that are returned vs exported and imported. There are some detail differences on data types (glue vs char, numeric vs integer), but I would say this should not be a problem.

### CODE REVIEW

File: R/run_stress_test.R [hunk 1]
```diff
@@ -4,7 +4,10 @@
 #' understand sensitivities of the scenarios, in which case the user may pass a
 #' vector of values to one (and only one) of the detail arguments. This will
 #' result in running the analysis multiple times in a row with the argument
-#' varied. NOTE: argument `asset_type` is not iterateable.
+#' varied.
+#' NOTE: argument `asset_type` is not iterateable.
+#' NOTE: if `return_results` is TRUE results will not be written to `output
```
Comment (jacobvjk) [inline R/run_stress_test.R:9]: Does this mean that output path becomes optional? Do we check if at least one is given? Or do we need both arguments regardless? Maybe something we need to be extra clear about  with the users

File: R/write_results.R [hunk 2]
```diff
@@ -1,127 +1,40 @@
...
-    ) %>%
-    dplyr::ungroup() %>%
-    dplyr::arrange(
-      .data$scenario_geography, .data$scenario_name, .data$investor_name,
-      .data$portfolio_name, .data$ald_sector, .data$year
-    )
+  results_list$stress_test_results_comp_el %>%
+    readr::write_csv(file.path(
+      output_path,
+      paste0("stress_test_results_", asset_type, "_comp_el_", iter_var, ".csv")
```
Comment (jacobvjk) [inline R/write_results.R:28]: why not always use `glue::glue()` for consistency?

**Review by jacobvjk — COMMENTED** [review]
some small comments

File: R/write_results.R [hunk 3]
```diff
@@ -1,127 +1,40 @@
...
-    report_all_duplicate_kinds(
-      composite_unique_cols = c(
-        "scenario_name", "scenario_geography", "investor_name", "portfolio_name",
-        "ald_sector", "term", sensitivity_analysis_vars
-      )
-    ) %>%
+  results_list$stress_test_results_sector_pd_changes_overall %>%
     readr::write_csv(file.path(
       output_path,
       paste0("stress_test_results_", asset_type, "_sector_pd_changes_overall_", iter_var, ".csv")
```
[the remark this answers is not in the document]
Comment (MirijaHa) [reply R/write_results.R:40]: done

File: R/write_results.R [hunk 4]
```diff
@@ -1,127 +1,40 @@
...
-    report_all_duplicate_kinds(
-      composite_unique_cols = c(
-        "scenario_name", "scenario_geography", "investor_name", "portfolio_name",
-        "ald_sector", "year", sensitivity_analysis_vars
-      )
-    ) %>%
+  results_list$stress_test_results_sector_pd_changes_annual %>%
     readr::write_csv(file.path(
       output_path,
       paste0("stress_test_results_", asset_type, "_sector_pd_changes_annual_", iter_var, ".csv")
```
[the remark this answers is not in the document]
Comment (MirijaHa) [reply R/write_results.R:34]: done

File: R/write_results.R (hunk 2 above)
Comment (MirijaHa) [reply R/write_results.R:28]: done

File: R/run_stress_test.R (hunk 1 above)
Comment (MirijaHa) [reply R/run_stress_test.R:9]: if return_results is TRUE output path does not need to be provided. Adjsuting.

Comment (MirijaHa) [reply R/run_stress_test.R:9]: actually,  I have to correct myself. even if result are returned we write log files to output paths, so path needs to be provided. Guess that couls be emphasised in vignette.

Comment (jacobvjk) [reply R/run_stress_test.R:9]: okay

**Review by jacobvjk — APPROVED** [review]
lgtm

**Review by maurolepore — COMMENTED** [review]
Note it's generally best to aim for a function to be type stable 

https://vctrs.r-lib.org/articles/stability.html

**Pull request merged**